types/protocol/protocol.go

```diff
@@ -0,0 +1,24 @@
+package protocol
+
+import (
+	"github.com/aquasecurity/tracee/types/detect"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+const (
+	TraceeEventSrc         = "tracee"
+	TraceeEventContentType = "tracee/event"
+)
+
+func ToEvent(e trace.Event) detect.Event {
+	origin := TraceeEventSrc + "/" + string(e.EventOrigin())
+	contentType := TraceeEventContentType + "-" + e.EventName
```

Comment:
Why do we add event name to the content type string? Is tracee-ebpf generating different kinds of events?

Reply:
First of all, tracee-ebpf can also output other types of payloads but that's besides the point.
As far as the signatures see the events they are different, since the selectors filter by the event's `Name`, by including them in the content type I can avoid unpacking them and keep it a bit more generic. 
In the future, I think we could move from event selectors to simple header comparisons between sigs and events, however I think including the event name would likely stay in that case.
Maybe we could add a different header for it?

Reply:
Usually ContentType indicates how to decode the Payload, e.g. distinguish Tracee-ebpf event sent as GOB or JSON from Kubernetes audit event sent as JSON or YAML.

If we concatenate ContentType with event name we're probably enriching original event with pre-calculated metadata that's only useful for processing events down the pipeline, but I'd not call it ContentType.

Reply:
Then sounds like a different header. I could use Customs to pass it, but again i'd like not to allocate maps so frequently, so I wonder if there is a hard coded header we could add here.

Reply:
Regarding ContentType its usually defined as serialisation format + API version + Embedded format. See, for example, how Kafka defines it https://docs.confluent.io/platform/current/kafka-rest/api.html

Reply:
But if tracee-rules is the only reader of the ContentType then we can maybe leave it with what you proposed in this PR. I don't have strong opinions until I see how we eventually use such ContentType header.

Reply:
I think we should open an issue about header rigor and stuff like that after we merge this. I do think there's room for improvement. But yes, atm tracee-rules is the only consumer of this ContentType.

Specifically, I think these headers will allow us, for example, to streamline the `SignatureSelector` mechanism significantly if not entirely remove it. When we get to that we should debate the rigor of this header.

Reply:
Agreed. For now we have ContentType that serves our purpose to avoid Go type casing in so many places. We can further decide what would be the best value for ContentType.